### ISSUE TITLE
Perm and Comb are calculated incorrectly

### ISSUE THREAD

**Issue by lwchkg** [tranleduy2000/ncalc#48]
Please check on Wolfram Mathworld for the definitions of these calculations:

Binomial coefficient: http://mathworld.wolfram.com/BinomialCoefficient.html
Factorial (you've done it right): http://mathworld.wolfram.com/Factorial.html

In particular, two things are wrong:

1. You shouldn't round off the numbers before you apply the factorial (or gamma) function.
1. When the result shows indeterminate, it should be actually zero sometimes. (The factorial/gamma function returns "indeterminate" because it reaches infinity. When a finite number is divided by that infinity, it becomes zero. And this is in line with the fact that nCr=0 when n, r are integers, r < 0 or r > n.)

**Comment by axkr** [comment]
Version 3.4.1 in IDE mode you can input the funtions like this
```
    Binomial(17,13)

    Factorial(0.75)
```
Can you give some examples of your calculations which go wrong.

**Comment by axkr** [comment]
You mean file  [CustomFunctions.java](https://github.com/tranleduy2000/ncalc/blob/master/app/src/main/java/com/duy/calculator/evaluator/CustomFunctions.java) where the definitions should be:

```
    Comb(n_, k_):=Binomial(n,k)
    Perm(n_, k_):=Gamma(n+1)/Gamma(k+1)
```

BTW in you screenshot the input `Comb(2.5,)` is parsed as `Comb(2.5, Null)`

**Comment by lwchkg** [comment]
Yes.

BTW, I don't understand why the second string is named `BINOMIAL`, which is supposed to be a synonym of "combination". I guess you should name that PERMUTATION instead.

Anyway, parsing `Comb(2.5,)` is parsed as `Comb(2.5, Null)` is okay to me. With invalid input anything other than crashing is okay. I gave you the screenshot just because it exposes the wrong formula.

**Comment by axkr** [comment]
Fixed with 3.4.3-beta
